Fix spring stiffness for BarReleases

IsakNaslundBh opened the pull request:
<!-- PLEASE ENSURE YOU REVIEW THE CONTENT OF EACH PR CAREFULLY, INCLUDING SUBSEQUENT COMMENTS BY YOURSELF OR OTHERS. -->
<!-- IN PARTICULAR PLEASE ENSURE THAT SENSITIVE OR INAPPROPRIATE INFORMATION IS NOT UPLOADED -->

 ### NOTE: Depends on 
<!-- Link to any additional PRs in other repos required for this PR to function -->
<!-- Delete if not required -->


 ### Issues addressed by this PR
<!-- Add reference(s) to issue(s) solved by this PR. Please use keyword Fixes/Closes as per https://help.github.com/articles/closing-issues-using-keywords/ -->

 Closes #393 

 <!-- Add short description of what has been fixed -->

Makes sure the spring stiffness is picked up by changing the check if a DOF is released from checking if identical to free to instead check if it is not fixed.

 ### Test files
<!-- Link to test files to validate the proposed changes -->

https://burohappold.sharepoint.com/:f:/s/BHoM/Evb6D2Mx5SlCldKS1XidhaUBl2Bi4gJ-lwtbV259ropqFg?e=0sPAc7

 ### Changelog
<!-- Text to go into changelog if applicable -->
<!-- Please see https://github.com/BHoM/documentation/wiki/changelog for guidelines -->


 ### Additional comments
<!-- As required -->

JosefTaylor reviewed (changes requested):
Needs similar change in ToBHoM in order to catch springs - see changes in SAP2000
Push works properly.

IsakNaslundBh commented:
thanks @JosefTaylor .

Have fixed now!

JosefTaylor commented on Etabs_Adapter/CRUD/Read/Bar.cs:
```diff
@@ -148,41 +148,41 @@ private List<Bar> ReadBar(List<string> ids = null)
-
-            BarRelease barRelease = new BarRelease() { StartRelease = startRelease, EndRelease = endRelease };
+            Constraint6DOF bhStartRelease = new Constraint6DOF();
+
```
Push uses XZY, pull uses XYZ- I am not sure which one is right! I think it is XZY.

JosefTaylor reviewed (changes requested):
axes are switched between pulling and pushing

IsakNaslundBh commented:
> axes are switched between pulling and pushing: Push uses XZY, pull uses XYZ- I am not sure which one is right! I think it is XZY.

You are right 🤦‍♂️ . Have fixed now, thanks for spotting.

Also added another section of the script to verify that the order is correct by looking at reactions, and as you stated, XZY looks to be correct. Really confusing with Etabs Left-hand rule with switched orientations between 1 and 2 compared to BHoM (and what I have seen in most other packages).

Anyhow, code updated and added a small method over the quite long statement of ?s and :s as I found that a bit hard to read if we look at this again :)

JosefTaylor reviewed (approved):
Works great!

JosefTaylor commented:
I have this transform matrix in SAP2000; is there a way we could use it to create releases?
`        public static TransformMatrix barLocalAxisToBHoM = new TransformMatrix()
        {
            Matrix = new double[4, 4]
            {
                { 1,  0,  0, 0 },
                { 0,  0, -1, 0 },
                { 0,  1,  0, 0 },
                { 0,  0,  0, 1 }
            }
        };`

IsakNaslundBh commented:
@BHoMBot check required

IsakNaslundBh commented:
@BHoMBot check copyright-compliance

IsakNaslundBh commented:
@BHoMBot check copyright-compliance
@BHoMBot check required
